commands/rez.js

```diff
@@ -0,0 +1,113 @@
+const enmap = require('enmap')
+const discord = require('discord.js')
+const rezClient = require('../modules/CampingReservationClient')
+
+exports.run = async (client, message, args, level) => {
+  const channel = message.channel
+  const author = message.author
+
+  const subcommand = args[0]
+  if(!subcommand) {
+    return sendHelp(channel)
+  }
+
+  const month = _getMonth(args[1])
+
+  switch(subcommand) {
+    case 'check':
+      await check(channel, month)
+      break
+    case 'watch':
+      await watch(client, channel, author, month)
+      break
+    case 'unwatch':
+      await unwatch(client, channel, author, month)
+      break
+    case 'watchlist':
+      await watchlist(client, channel, author)
+      break
+    default:
+      return sendHelp(channel)
+  }
+}
+
+async function check(channel, month) {
+  const reservations = await rezClient.getReservations(month)
+
+  if(reservations.size) {
+    channel.send(`\`\`\`
+${rezClient.getAsTable(reservations)}
+\`\`\``);
+  } else {
+    channel.send(`No reservations for month: ${month}`)
+  }
+}
+
+async function watch(client, channel, discordUser, month) {
+  const key = discordUser.id;
+  const userData = _getUserData(client, key)
+
+  if (!userData.monthsToCheck.includes(month)) {
+    client.rezData.push(key, month, "monthsToCheck", false)
+  }
+
+  _sendWatchList(client, channel, key)
+}
+
+async function unwatch(client, channel, discordUser, month) {
+  const key = discordUser.id;
+  const userData = _getUserData(client, key)
+
+  if (userData.monthsToCheck.includes(month)) {
+    client.rezData.remove(key, month, 'monthsToCheck')
+  }
+
+  _sendWatchList(client, channel, key)
+}
+
+async function watchlist(client, channel, discordUser) {
+  const key = discordUser.id;
+  _sendWatchList(client, channel, discordUser.id)
+}
+
+function _getUserData(client, userId) {
```

Comment:
minor: if this is storing by user id, it will be persistent across all the discord guilds right? I think better key would be guildId+userId

Reply:
okayyyy

Reply:
Hm when you DM the bot, theres no guildId, any workarounds for that?

Reply:
maybe I should make this bot DM-only so guilds don't matter

Reply:
that works too